### ISSUE TITLE
Large mobs can spawn in small cavities

### ISSUE THREAD

**Issue by Wuzzy2** [Wuzzy2/MineClone2-Bugs#133]
Large mobs (such as the huge slime) can spawn in cavities way too big for them. This has to change.

Also, killing the huge slime has good chances for the new slime to end up inside walls (no collision check).

**Comment by Bu-Gee** [comment]
Just in case you haven't seen it, I've had an Enderman show up in a space that was vertically too short for it.  If it isn't the same issue, I can open a separate bug report for it.

**Comment by Bu-Gee** [comment]
Here is a video of it: https://youtu.be/tGzaafHveNc?t=16m54s

**Comment by WSDguy2014** [comment]
I got a same thing in subgame version 0.15.0
![screenshot_20170310_232236](https://cloud.githubusercontent.com/assets/13905392/23813688/adf03582-05e8-11e7-8491-9114030949c3.png)

But i hit a big slime for example, few seconds later, big slime starts sinking inside of blocks.
![screenshot_20170310_232301](https://cloud.githubusercontent.com/assets/13905392/23813711/c81dee40-05e8-11e7-88be-63b7dfd4f2ca.png)

**Comment by kwadroke** [comment]
I've seen this as well.

**Comment by leorockway** [comment]
I found a spider spawner and the place had only one opening. A spider got out through that opening and I attacked it. Several other spiders came to attack me from the room with the spawner, but they went through the wall. I believe this might be related to this bug, where big monsters can occupy spaces that they shouldn't.

**Comment by Wuzzy2** [comment]
Yeah, the Mobs Redo spawning algorithm needs a major overhaul. The problem is that apparently, Mobs Redo does not check at all whether the mob has enough space to spawn; at least it is not customizable.
I guess I'll fix most of the mob bugs by directly contributing to Mobs Redo first, then pulling it back to MineClone 2.
This is one of these bugs.
